### PR TITLE
UNR-704: Remove actor log spam, clean up deletion flow

### PR DESCRIPTION
**Pull request by davedolben**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
- Don't attempt to remove entities from registry or net driver twice on authoritative worker
- (related fix) Don't attempt to delete actors in PIE on channel close unless the world is shutting down

#### Tests
Tested deletion flow and verified log output manually on a modified branch of starter project.

#### Documentation
n/a

#### Primary reviewers
@Vatyx @girayimprobable

### CODE REVIEW

File: SpatialGDK/Source/Private/Utils/EntityRegistry.cpp [hunk 1]
```diff
@@ -27,11 +27,32 @@ void UEntityRegistry::RemoveFromRegistry(const Worker_EntityId& EntityId)
...
+			ActorToEntityId.Remove(Actor);
+		}
+		else
+		{
+			UE_LOG(LogEntityRegistry, Warning, TEXT("Tried to remove actor %s from registry but it wasn't there"), *Actor->GetFullName());
+		}
+	}
+	else
+	{
+		UE_LOG(LogEntityRegistry, Warning, TEXT("Couldn't remove Actor from registry: Actor == nullptr"));
```
Comment (girayimprobable) [inline SpatialGDK/Source/Private/Utils/EntityRegistry.cpp]: Output the entity id for debugging please.

File: SpatialGDK/Source/Private/Utils/EntityRegistry.cpp [hunk 2]
```diff
@@ -27,11 +27,32 @@ void UEntityRegistry::RemoveFromRegistry(const Worker_EntityId& EntityId)
...
+	}
 
 	if (EntityId != -1)
```
Comment (danielimprobable) [inline SpatialGDK/Source/Private/Utils/EntityRegistry.cpp]: Do we have a constant for invalid entity id?

Comment (davedolben) [reply SpatialGDK/Source/Private/Utils/EntityRegistry.cpp]: Yes, but it doesn't actually cover all invalid entity IDs (must be > 0). Changed to check for that instead

File: SpatialGDK/Source/Private/Interop/SpatialReceiver.cpp [hunk 3]
```diff
@@ -400,9 +400,10 @@ void USpatialReceiver::RemoveActor(Worker_EntityId EntityId)
 	{
 		if (USpatialActorChannel* ActorChannel = NetDriver->GetActorChannelByEntityId(EntityId))
 		{
+			UE_LOG(LogSpatialReceiver, Warning, TEXT("RemoveActor: actor for entity %lld was already deleted (likely on the authoritative worker) but still has an open actor channel"), EntityId);
```
Comment (joshuahuburn) [inline SpatialGDK/Source/Private/Interop/SpatialReceiver.cpp]: Nit: Full stop.

File: SpatialGDK/Source/Private/Utils/EntityRegistry.cpp [hunk 4]
```diff
@@ -27,15 +27,37 @@ void UEntityRegistry::RemoveFromRegistry(const Worker_EntityId& EntityId)
...
-	ActorToEntityId.Remove(Actor);
+	if (Actor)
+	{
+		if (ActorToEntityId.Contains(Actor))
+		{
+			ActorToEntityId.Remove(Actor);
+		}
+		else
+		{
+			UE_LOG(LogEntityRegistry, Warning, TEXT("Tried to remove actor %s (entity %lld) from registry but it wasn't there"),
```
Comment (joshuahuburn) [inline SpatialGDK/Source/Private/Utils/EntityRegistry.cpp]: Nits: Full stops.

File: SpatialGDK/Source/Private/Interop/SpatialReceiver.cpp (hunk 3 above)
Comment (davedolben) [reply SpatialGDK/Source/Private/Interop/SpatialReceiver.cpp]: Is this a British thing?

**Pull request merged**